frontend/src/app/modules/matches/matches-computations.js

```diff
@@ -68,3 +79,71 @@ export const computeWinRatio = (playerId, playerMatches) => {
   const wonMatchesCount = playerMatches.filter(match => didPlayerWin(playerId, match)).length
   return (playerMatches.length > 0) ? (wonMatchesCount / playerMatches.length) : 0
 }
+
+export const computeWins = (playerId, playerMatches) =>
+  playerMatches.filter(match => didPlayerWin(playerId, match)).length
+
+const findWithComparator = (comparator, arr, getField, baseScore) =>
+  arr.reduce((acc, node) =>
+    comparator(acc.score, getField(node))
+      ? acc
+      : { value: node.value, score: getField(node) }
+  , { score: baseScore })
+
+export const findMin = (arr, getField = node => node.score) =>
+  findWithComparator((e1, e2) => e1 <= e2, arr, getField, Infinity)
+
+export const findMax = (arr, getField = node => node.score) =>
+  findWithComparator((e1, e2) => e1 >= e2, arr, getField, -Infinity)
+
+export const computeDays = matchChanges => {
+  const daysMap = new Map()
+
+  for (const match of matchChanges) {
+    const day = match.date.toLocaleDateString()
+    const score = Number(match.ratingChangeString)
+    daysMap.set(day, !daysMap.has(day)
```

Comment:
it's common practice to minimize use of the negations as it's easier to read... if you just turn this condition as `daysMap.has(day)? A : B` it could be easier to read than `!daysMap.has(day)? B : A`

Reply:
otherwise I like the use of Map here :+1:

Reply:
De-negated [here](https://github.com/salsita/foosball-rating/pull/114/commits/54023373857edbdd2d03d427c0b799b98214f08c#diff-304dca28d92d292252e86b7e83fc1569)